runtime/memory.c

```diff
@@ -464,29 +464,17 @@ CAMLexport color_t caml_allocation_color (void *hp)
 }
 
 Caml_inline value caml_alloc_shr_aux (mlsize_t wosize, tag_t tag, int track,
-                                      int raise_oom, uintnat profinfo)
+                                      uintnat profinfo)
 {
   header_t *hp;
   value *new_block;
 
-  if (wosize > Max_wosize) {
-    if (raise_oom)
-      caml_raise_out_of_memory ();
-    else
-      return 0;
-  }
+  if (wosize > Max_wosize) return 0;
   CAML_EV_ALLOC(wosize);
   hp = caml_fl_allocate (wosize);
   if (hp == NULL){
     new_block = expand_heap (wosize);
-    if (new_block == NULL) {
-      if (!raise_oom)
-        return 0;
-      else if (Caml_state->in_minor_collection)
-        caml_fatal_error ("out of memory");
-      else
-        caml_raise_out_of_memory ();
-    }
+    if (new_block == NULL) return 0;
```

Comment:
There is a small change in error-handling in this patch. Before, `wosize > Max_wosize` would always raise an oom, while `expand_heap` failure would fatal-error during minor collections and raise oom otherwise. Now, both failure conditions would fatal-error during minor conditions. I suppose that you believe that the change is harmless?

(I am not sure when it can be the case that `wosize > Max_wosize` is passed to this function, from a distance this looks like a programming error?)

Reply:
> I suppose that you believe that the change is harmless?

Yesterday, I'd more-or-less convinced myself that the case was impossible, because during minor collection the only allocations are promotions which are always much less than Max_wosize. However, I think you can in fact hit this case, by doing a very large allocation via `caml_alloc_shr` from user code in a custom block finaliser called during minor GC.

If you do that, though, you'll end up raising an exception from the middle of minor GC, which is bad. The minor heap is in an invalid state, and you're likely to segfault during exception handling.

So: I think the difference is only observable in extremely weird / broken code, and in this case the new behaviour is a mild improvement.